src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -8062,7 +8068,7 @@ private bool IsPossibleNewExpression()
 
                 // class, struct, enum, interface keywords, but also other modifiers that are not allowed after 
                 // partial keyword but start class declaration, so we can assume the user just swapped them.
-                if (IsPossibleStartOfTypeDeclaration(PeekToken(2).Kind))
+                if (IsTypeModifierOrTypeKeyword(PeekToken(2).Kind))
```

Comment:
2 things to note:
1. this helper is only called for ParseMemberDeclarationOrStatement. i.e., top-level declarations that could either be a member or a statement.
2. The intent here seems to be to ask: are we looking at `new partial` followed by something which indicates this is part of a member declaration. `new partial [` isn't going to parse as a single member declaration. `new partial MODIFIER` is not valid, but we still try to keep going and parse as a member declaration in such a case

This probably doesn't behave well when `new partial` is followed by a contextual keyword like `new partial file ...`. That is still an error case at this point, though, so it just means our error recovery won't be the best. Eventually I'm sure we'll do the work to relax partial modifier ordering, at which point we might be able to refactor a bunch of this code, and maybe do something like "parse the modifiers and see if we had a partial modifier. if it's not the last modifier then do a LangVersion check".